Cargo.toml

```diff
@@ -18,6 +18,76 @@ members = [
 ]
 resolver = "2"
 
+[workspace.dependencies]
+ark-algebra-test-templates = "0.3.0"
+ark-bn254 = { version = "0.3.0" }
+ark-ec = { version = "0.3.0", features = ["parallel"] }
```

Comment:
I'm not sure if declaring features in `Cargo.toml` is a good idea -- if I understand correctly, they are additive with features requested in any of the sub-packages (e.g. kimchi). I guess it's fine for these libraries since we pretty much want these features (parallel, asm) in all of them?

Reply:
the way the workspace resolver works, if any dep in the workspace requests a feature, every use of the crate will have that feature. this centralizing of features into the toplevel config basically encodes the status quo that was represented in the lockfile

Reply:
one detail i'm not sure about is how this interacts with cargo publish: if certain crates truly are agnostic towards the feature maybe we should override the workspace features so that the computed package manifest doesn't unnecessarily include the features even when its usage in a typical workspace will have them anyway